packages/financial-templates-lib/src/clients/OptimisticOracleClient.ts

```diff
@@ -67,28 +102,40 @@ export class OptimisticOracleClient {
   }
 
   // Returns an array of Price Requests that have no proposals yet.
-  public getUnproposedPriceRequests(): RequestPrice[] {
+  public getUnproposedPriceRequests(): RequestPriceReturnValues[] {
     return this.unproposedPriceRequests;
   }
 
   // Returns an array of Price Proposals that have not been disputed.
-  public getUndisputedProposals(): ProposePrice[] {
+  public getUndisputedProposals(): ProposePriceReturnValues[] {
     return this.undisputedProposals;
   }
 
   // Returns an array of expired Price Proposals that can be settled and that involved
   // the caller as the proposer
-  public getSettleableProposals(caller: string): ProposePrice[] {
-    return this.expiredProposals.filter((event) => {
-      return event.proposer === caller;
-    });
+  public getSettleableProposals(caller: string): ProposePriceReturnValues[] {
+    if (this.oracleType === "SkinnyOptimisticOracle") {
```

Comment:
having this not be string comparison would be nice.

Reply:
How can we check type conditionally?

Reply:
Actually, enums are easy. I changed to enums